lib/semian/pid_controller.rb

```diff
@@ -87,11 +87,7 @@ def update
         # Update rejection rate (clamped between 0 and 1)
         @rejection_rate = new_rejection_rate.clamp(0.0, 1.0)
 
-        # Anti-windup: back out the integral accumulation if output was saturated
-        if new_rejection_rate != @rejection_rate
-          # Output was clamped, reverse the integral accumulation
-          @integral -= @last_p_value * dt
-        end
+        @integral = @integral.clamp(-10.0, 10.0)
```

Comment:
I'm not really sure what was wrong with this clamping method, but looking at the integral behaviour with it, it was very weird, and never settling back to 0.

A simple clamp of -10,10 seems to work and make sure it does go back to 0 eventually